lib/Infer/AliveDriver.cpp

```diff
@@ -0,0 +1,406 @@
+#include <souper/Infer/AliveDriver.h>
+
+#include <string_view>
+#include <memory>
+#include <set>
+
+#include "llvm/Support/raw_ostream.h"
+
+#include "alive2/ir/instr.h"
+#include "alive2/ir/function.h"
+#include "alive2/smt/smt.h"
+#include "alive2/smt/solver.h"
+#include "alive2/ir/state.h"
+#include "alive2/smt/expr.h"
+#include "alive2/util/errors.h"
+#include "alive2/util/symexec.h"
+#include "alive2/tools/transform.h"
+#include "alive2/smt/ctx.h"
+#include <z3.h>
+
+namespace {
+class FunctionBuilder {
+public:
+  FunctionBuilder(IR::Function& F_) : F(F_) {}
+
+  template <typename A, typename B, typename ...Others>
+  IR::Value *BinOp(IR::Type &t, std::string name, A a, B b,Others... others) {
+    return Append
+        (std::make_unique<IR::BinOp>
+         (t, std::move(name), *ToValue(t, a), *ToValue(t, b), others...));
```

Comment:
more odd indentation-- do a search for tabs in your patch and eliminate all of them